test/e2e/utils.go

```diff
@@ -728,3 +741,120 @@ func getEnvAndExpectNoError(key string) string {
 	utilruntime.Must(err)
 	return val
 }
+
+// newTestHelperJob returns the K8s Job for given commands to be executed inside k8s cluster.
+// This test helper job can be used to validate test cases from inside the k8s cluster by executing the bash scripts.
+func newTestHelperJob(jobName string, podSpec *corev1.PodSpec) *batchv1.Job {
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: *podSpec,
+			},
+			BackoffLimit: pointer.Int32(0),
+		},
+	}
+}
+
+// etcdZeroDownTimeValidatorJob returns k8s job which ensures
+// Etcd cluster zero down time by continuously checking etcd cluster health.
+// This job fails once health check fails and associated pod results in error status.
+func etcdZeroDownTimeValidatorJob(etcdSvc, testName string, tls *v1alpha1.TLSConfig) *batchv1.Job {
+	return newTestHelperJob(
+		"etcd-zero-down-time-validator-"+testName,
+		&corev1.PodSpec{
+			Volumes: []v1.Volume{
+				{
+					Name: "client-url-ca-etcd",
+					VolumeSource: v1.VolumeSource{
+						Secret: &v1.SecretVolumeSource{
+							SecretName:  tls.TLSCASecretRef.Name,
+							DefaultMode: pointer.Int32(420),
+						},
+					},
+				},
+				{
+					Name: "client-url-etcd-server-tls",
+					VolumeSource: v1.VolumeSource{
+						Secret: &v1.SecretVolumeSource{
+							SecretName:  tls.ClientTLSSecretRef.Name,
+							DefaultMode: pointer.Int32(420),
+						},
+					},
+				},
+				{
+					Name: "client-url-etcd-client-tls",
+					VolumeSource: v1.VolumeSource{
+						Secret: &v1.SecretVolumeSource{
+							SecretName:  tls.ServerTLSSecretRef.Name,
+							DefaultMode: pointer.Int32(420),
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:    "etcd-zero-down-time-validator-" + testName,
+					Image:   "alpine/curl",
+					Command: []string{"/bin/sh"},
+					//To avoid flakiness, consider downtime when curl fails consecutively back-to-back.
+					Args: []string{"-ec",
+						"echo '" +
+							"failed=0 ; threshold=2 ; " +
+							"while [ $failed -lt $threshold ] ; do  " +
```

Comment:
> //To avoid flakiness, consider downtime when curl fails consecutively back-to-back.

Is consecutively failure of `curl` determine by `threshold=2` ?
And how it helps in removing the flakiness ?

Reply:
Zero downtime validator Job which continuously `curls` **Etcd cluster health** before triggering our test case.
let say when we trigger rolling update `etcd-03` pod will be restarted  at the same moment curl may point to IP of  `etcd-3 pod` because we are using `etcd-client` svc (This happens in rare case).